### PR TITLE
Create in-repo plugin project for pipeline editor.

### PR DESCRIPTION
**Pull request by sophistifunk**
This PR exists mostly as an anchor and to run the build in CI. I'll close it and make a new one before this goes into master.

### CODE REVIEW

File: blueocean-pipeline-editor/pom.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>blueocean-parent</artifactId>
+        <groupId>io.jenkins.blueocean</groupId>
+        <version>1.0-alpha-7-SNAPSHOT</version>
```
Comment (imeredith) [inline blueocean-pipeline-editor/pom.xml]: This needs to be 1.0-alpha-8-SNAPSHOT

**Comment by michaelneale** [comment]
` <version>1.0-alpha-8-SNAPSHOT</version>` shoudl ve verison of editor

**Comment by michaelneale** [comment]
LGTM - do you want to leave this as a WIP PR or you want to merge it and keep dev in a branch for later?

File: blueocean-pipeline-editor/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        "space-before-function-paren": [2, {
+            "anonymous": "always",
+            "named": "never"
+        }],
+        "strict": [2, "global"]
+    },
+    "plugins": [
+        "react"
+    ]
+}
```
Comment (tfennelly) [inline blueocean-pipeline-editor/.eslintrc]: Another different `.eslintrc` file 😿

File: blueocean-pipeline-editor/src/main/js/components/editor/EditorPipelineGraph.jsx [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+type ConnectionInfo = [NodeInfo, NodeInfo];
+
+type LabelInfo = {
+    //key: string,
+    x: number,
+    y: number,
+    text: string,
+    node: NodeInfo,
+    stage?: StageInfo
+};
```
Comment (tfennelly) [inline blueocean-pipeline-editor/src/main/js/components/editor/EditorPipelineGraph.jsx]: All this typing stuff is nice !!!

File: blueocean-pipeline-editor/src/main/js/jenkins-js-extension.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Extensions in this plugin
+# NB: "component" currently maps to modules, not "symbols" so make sure to "export default"
+# WARNING: If you change this you'll have to change io.jenkins.blueocean.jsextensions.JenkinsJSExtensionsTest as well :(
```
Comment (tfennelly) [inline blueocean-pipeline-editor/src/main/js/jenkins-js-extension.yaml]: A big dirty warning is given if there's no default export, so I don't think there's a need for that comment anymore.

**Comment by tfennelly** [comment]
Looks good insofar as I get it.

File: blueocean-pipeline-editor/.eslintrc (hunk 2 above)
Comment (sophistifunk) [inline blueocean-pipeline-editor/.eslintrc]: What part of "style complaints to /dev/null" was I vague about? :D

File: blueocean-pipeline-editor/src/main/js/components/editor/EditorPipelineGraph.jsx (hunk 3 above)
Comment (sophistifunk) [inline blueocean-pipeline-editor/src/main/js/components/editor/EditorPipelineGraph.jsx]: Yes, I find it helps me more than it annoys me, which is basically as good as current type systems get

Comment (michaelneale) [inline blueocean-pipeline-editor/src/main/js/components/editor/EditorPipelineGraph.jsx]: \o/

**Comment by michaelneale** [comment]
thanks @tfennelly - will leave this is a WIP. @sophistifunk if you want you can close request and keep branch alive and open again when you want more review. OR, leave this as "work in progress" and just add the "needs-review" tag when there is more to look at?

File: blueocean-pipeline-editor/.eslintrc (hunk 2 above)
Comment (tfennelly) [inline blueocean-pipeline-editor/.eslintrc]: Ah sorry. I was just referring to the existence of another one of these files (containing a different set of rules). I never comment on "style" 😃  ... that's the advantage of a consistent set of automated rules ... people shouldn't have to comment.

**Comment by sophistifunk** [comment]
The cheese is old and mouldy. Where is the bathroom please?

**Pull request closed without merging**